db/range_del_aggregator.cc

```diff
@@ -365,29 +271,208 @@ Status RangeDelAggregator::AddTombstone(RangeTombstone tombstone) {
       if (new_to_old_end_cmp < 0) {
         ++new_range_dels_iter;
       } else if (new_to_old_end_cmp > 0) {
-        ++tombstone_map_iter;
+        ++rep_iter;
         untermed_seq = kMaxSequenceNumber;
       } else {
         ++new_range_dels_iter;
-        ++tombstone_map_iter;
+        ++rep_iter;
         untermed_seq = kMaxSequenceNumber;
       }
     }
     while (new_range_dels_iter != new_range_dels.end()) {
-      tombstone_map.emplace(
-          new_range_dels_iter->start_key_,
-          RangeTombstone(Slice(), Slice(), new_range_dels_iter->seq_));
+      rep_.emplace(new_range_dels_iter->start_key_,
+                   RangeTombstone(Slice(), Slice(), new_range_dels_iter->seq_));
       ++new_range_dels_iter;
     }
+  }
+
+  size_t Size() const { return rep_.size() - 1; }
+
+  void InvalidatePosition() { iter_ = rep_.end(); }
+
+  typedef std::multimap<Slice, RangeTombstone, stl_wrappers::LessOfComparator>
+      Rep;
+
+  class Iterator : public RangeDelIterator {
```

Comment:
Similar to other comment: can `Iterator` and `Rep` be private?

Reply:
Done.